assets/js/components/notifications/EnableAutoUpdateBannerNotification.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * EnableAutoUpdateBannerNotification component.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useCallback, useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useSelect, useDispatch } from 'googlesitekit-data';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import useViewContext from '../../hooks/useViewContext';
+import SimpleNotification from '../../googlesitekit/notifications/components/layout/SimpleNotification';
+import Description from '../../googlesitekit/notifications/components/common/Description';
+import ActionsCTALinkDismiss from '../../googlesitekit/notifications/components/common/ActionsCTALinkDismiss';
+import Dismiss from '../../googlesitekit/notifications/components/common/Dismiss';
+
+export const ENABLE_AUTO_UPDATES_BANNER_SLUG = 'auto-update-cta';
+
+export default function EnableAutoUpdateBannerNotification( {
+	id,
+	Notification,
+} ) {
+	const viewContext = useViewContext();
+
+	const siteKitAutoUpdatesEnabled = useSelect( ( select ) =>
+		select( CORE_SITE ).getSiteKitAutoUpdatesEnabled()
+	);
+	const enableAutoUpdateError = useSelect( ( select ) =>
+		select( CORE_SITE ).getErrorForAction( 'enableAutoUpdate', [] )
+	);
+
+	const { enableAutoUpdate } = useDispatch( CORE_SITE );
+	const ctaActivate = useCallback( async () => {
+		await enableAutoUpdate();
+	}, [ enableAutoUpdate ] );
+
+	const [ enabledViaCTA, setEnabledViaCTA ] = useState( false );
+
+	// If auto-updates were enabled via this CTA banner, then set the state to
+	// true to render the success banner variation.
+	useEffect( () => {
+		if ( enabledViaCTA === false && siteKitAutoUpdatesEnabled === true ) {
+			setEnabledViaCTA( true );
+		}
+	}, [ enabledViaCTA, siteKitAutoUpdatesEnabled ] );
```

Comment:
I believe `setEnabledViaCTA` should be also added to the dependencies array, shouldn't it?